### PR TITLE
Update: Template Browser Filtering

### PR DESCRIPTION
**Pull request by jennsun**
https://app.clickup.com/t/14kqd0a

Update the collections page on the SDK Demo site so that video templates have more filtering options: duration and aspect ratio. The text color of the applied filters is now the Waymark blue color. Multiple filters are allowed, and the duration and aspect ratio filters can be toggled on and off. All Templates is now the default collection if no other category filter is chosen. The categories are now listed in alphabetical order on the sidebar. Also fixed Account Creation issue and grid issue (in the past, titles that were too long prevented the template grid from displaying 2 per row).

### CODE REVIEW

File: example_site/src/components/Collections.js [hunk 1]
```diff
@@ -31,45 +32,124 @@ function Template({ template }) {
   );
 }
 
+function DurationFilter({
+  filter,
+  templateFilter,
+  setTemplateFilter,
+}) {
+  const [buttonStatus, setButtonStatus] = useState(true);
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: What exactly is `buttonStatus` controlling?

File: example_site/src/components/Collections.js [hunk 2]
```diff
@@ -31,45 +32,124 @@ function Template({ template }) {
...
+    }
+  }
+
+  return (
+    <div className="category-name">
+      <button 
+        className="filter-name"
+        onClick={() => {
+          onSelectFilter(filter)
+          setButtonStatus(!buttonStatus)
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: Is there a reason that this `setButtonStatus` call is not included in `onSelectFilter`?

File: example_site/src/components/Collections.js [hunk 3]
```diff
@@ -31,45 +32,124 @@ function Template({ template }) {
...
+      }}>
+        <div style={{'color': buttonColor}}>
+          {filter.displayName} 
+        </div>
+      </button>
+    </div>
+  );
+}
+
+function AspectRatioFilter({
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: This and the `DurationFilter` component are identical except for the fact that they're dealing w/ aspect ratio vs. duration. Let's make one reusable component!

File: example_site/src/components/Collections.js [hunk 4]
```diff
@@ -96,7 +176,7 @@ function CollectionTemplates({
 
 export default function Collections() {
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: I think we should rename this component and file now that it's redone to resemble our template browser - what do you think about `TemplateBrowser`? (very creative I know)

File: example_site/src/components/Collections.js [hunk 5]
```diff
@@ -31,45 +32,124 @@ function Template({ template }) {
...
 }) {
-  const { waymarkInstance, addTemplates } = useAppContext();
+  const buttonColor = selectedCollection === collection ? theBlue : 'black';
 
   const onClick = (collection) => {
     setSelectedCollection(collection);
   };
 
+  if (!selectedCollection) {
+    setSelectedCollection(collections.find((collection) => (
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: What do you think of keeping this logic in the top level component and always passing in a selected collection? The only time there should not be one is when the component loads for the first time, so we should be able to put this logic in selected collection state initialization.

File: example_site/src/components/Collections.js [hunk 6]
```diff
@@ -124,23 +211,42 @@ export default function Collections() {
...
+                <AspectRatioFilter
+                  filter={filter}
+                  templateFilter={templateFilter}
+                  setTemplateFilter={setTemplateFilter}
+                />
+              ))}
+
               <h2>Categories</h2>
               {collections.map((collection) => (
                 <CollectionNames
```
Comment (simanthastew) [inline example_site/src/components/Collections.js]: While we're at it, let's rename this to `CollectionFilter` to stay consistent with the other components.

**Review by simanthastew — CHANGES_REQUESTED** [review]
Looking really good! A few small things to clean up, and then I think we can make one re-usable filter component.

File: example_site/src/components/Collections.js (hunk 1 above)
Comment (jennsun) [reply example_site/src/components/Collections.js]: Whether the filter is applied or not - I agree this is a vague variable name. Renamed to filterApplied, setFilterApplied.

File: example_site/src/components/Collections.js (hunk 2 above)
Comment (jennsun) [reply example_site/src/components/Collections.js]: Nope, put into onSelectFilter!

File: example_site/src/components/TemplateBrowser.js [hunk 7]
```diff
@@ -32,23 +32,26 @@ function Template({ template }) {
...
-function DurationFilter({
+function Filter({
   filter,
+  filterKey,
   templateFilter,
   setTemplateFilter,
 }) {
-  const [buttonStatus, setButtonStatus] = useState(true);
-  const buttonColor = templateFilter.duration === filter.value ? theBlue : 'black';
+  const [filterIsApplied, setFilterIsApplied] = useState(true);
```
Comment (simanthastew) [inline example_site/src/components/TemplateBrowser.js]: This is super nitpicky and you had no way of knowing, but we typically start boolean values w/ is, should, or has, so let's change this to `[isFilterApplied, setIsFilterApplied`.

File: example_site/src/components/TemplateBrowser.js [hunk 8]
```diff
@@ -60,71 +63,26 @@ function DurationFilter({
...
 
-function CollectionNames({
-  collections,
+function CollectionFilter({
   collection,
   selectedCollection,
   setSelectedCollection,
 }) {
-  const buttonColor = selectedCollection === collection ? theBlue : 'black';
+  const buttonColor = selectedCollection===collection ? theBlue : 'black';
```
Comment (simanthastew) [inline example_site/src/components/TemplateBrowser.js]: You should add the spaces back in!

**Review by simanthastew — CHANGES_REQUESTED** [review]
Two very small changes then you're good to merge!

**Pull request merged**